handler/websocket.go

```diff
@@ -57,56 +58,117 @@ func connectWs(exec graphql.ExecutableSchema, w http.ResponseWriter, r *http.Req
 		active: map[string]context.CancelFunc{},
 		exec:   exec,
 		conn:   ws,
-		ctx:    r.Context(),
 		cfg:    cfg,
 	}
 
-	if !conn.init() {
+	initMessage, ok := conn.init()
+	if !ok {
 		return
 	}
 
-	conn.run()
+	// When the websocket connection is initialized, and a onConnect
+	// function is defined, then we should pass the payload data up to
+	// the handler.
+	connectionParams := map[string]interface{}{}
+	if initMessage.Payload != nil {
+		if err := json.Unmarshal(initMessage.Payload, &connectionParams); err != nil {
+			conn.sendConnectionError("invalid json")
+			return
+		}
+	}
+
+	next := func(ctx context.Context, params map[string]interface{}) error {
+		return conn.run(ctx)
+	}
+
+	if err := cfg.onConnectHook(r.Context(), connectionParams, next); err != nil {
+		// TODO: handle the error somehow?
```

Comment:
please handle this, you can see an example a few lines above of returning an error to the client.